File: backend/src/domain/character/character_test.go
package character

import (
	"testing"
)

const ULID = "01AN4Z07BY79KA1307SR9X4MV3"

func TestErrorWhenWrongUlid(t *testing.T) {
	_, err := NewCharacter("asd", "Chindasvinto", 1)
	if err == nil {
		t.Errorf("Should return error when wrong ULID")
	}
}

func TestValues(t *testing.T) {
	sut, _ := dummyCharacter()
	if sut.Id() != ULID {
		t.Errorf("Not matching ids")
	}
	if sut.Name() != "Chindasvinto" {
		t.Errorf("Character name not valid!!!")
	}
  if sut.Level() != 2 {
		t.Errorf("Level not valid!!!")
  }
}

func dummyCharacter() (Character, error) {
	return NewCharacter(ULID, "Chindasvinto", 300)
}

File: frontend/tests/Domain/Party/Character/Character.test.ts
import { Character } from '@/Domain/Party/Character/Character'
import { isValid } from 'ulidx'
import { describe, expect, test } from 'vitest'

describe('Testing CreatePartyCommandHandler', () => {
  test('It should return proper values', () => {
    const sut = Character.fromValues('01HWXKZP2RSNS1J9PBK1V3P3X7', 'Chindasvinto', 120)
    expect(sut.ulid()).toBe('01HWXKZP2RSNS1J9PBK1V3P3X7')
    expect(sut.name()).toBe('Chindasvinto')
    expect(sut.xp()).toBe(120)
  })
  test('It should return proper values when no ulid given', () => {
    const sut = Character.withNameAndXp('Chindasvinto', 120)
    expect(isValid(sut.ulid())).toBe(true)
    expect(sut.name()).toBe('Chindasvinto')
    expect(sut.xp()).toBe(120)
  })
  test('It should throw error when name empty', () => {
    expect(() => {
      Character.fromValues('01HWXKZP2RSNS1J9PBK1V3P3X7', '', 120)
    }).toThrow(RangeError)
  })
  test('It should throw error when wrong ulid', () => {
    expect(() => {
      Character.fromValues('asd', 'Chindasvinto', 120)
    }).toThrow(RangeError)
  })
  test('It should throw error when experience points less than zero', () => {
    expect(() => {
      Character.fromValues('01HWXM7G2RXYW2SCNJJ38171P9', 'Chindasvinto', -1)
    }).toThrow(RangeError)
  })
})
